### ISSUE TITLE
Catch-Filter results in loop when two(?) lured pokemon is detected

### ISSUE THREAD

**Issue by akanezumi** [PokemonGoF/PokemonGo-Bot#4153]
### Expected Behavior
Filtering for some Pokemons to not being captured should totally ignore that kind/type of Pokemon.

### Actual Behavior
Enabling a catch filter for some pokemons results in a loop and a stuck bot when two(?) lured pokemons are detected.

### Your FULL config.json (remove your username, password, gmapkey and any other private info)
```
{
    "auth_service": "ptc",
    "encrypt_location": "",
    "heartbeat_threshold": 10,
    "tasks": [
      {
        "type": "HandleSoftBan"
      },
      {
        "type": "SleepSchedule",
        "config": {
          "enabled": false,
          "time": "22:54",
          "duration":"7:46",
          "time_random_offset": "00:24",
          "duration_random_offset": "00:43"
        }
      },
      {
        "type": "SleepSchedule",
        "config": {
          "enabled": false,
          "time": "12:00",
          "duration":"1:23",
          "time_random_offset": "00:10",
          "duration_random_offset": "00:20"
        }
      },
      {
        "type": "CollectLevelUpReward"
      },
      {
        "type": "IncubateEggs",
        "config": {
          "longer_eggs_first": false
        }
      },
      {
        "type": "UpdateLiveStats",
        "config": {
          "enabled": true,
          "min_interval": 300,
          "stats": ["uptime", "stardust_earned", "xp_earned", "xp_per_hour", "stops_visited"],
          "terminal_log": true,
          "terminal_title": true
        }
      },
      {
        "type": "TransferPokemon"
      },
      {
        "type": "NicknamePokemon",
        "config": {
          "enabled": false,
          "nickname_template": "{name}_{iv_ads_hex}"
        }
      },
      {
        "type": "EvolvePokemon",
        "config": {
            "evolve_all": "none",
            "first_evolve_by": "iv",
            "evolve_above_cp": 700,
            "evolve_above_iv": 0.8,
            "logic": "or",
            "evolve_speed": 20,
            "use_lucky_egg": false
        }
      },
      {
        "type": "RecycleItems",
        "config": {
          "min_empty_space": 15,
          "max_balls_keep": 152,
          "max_potions_keep": 49,
          "max_berries_keep": 68,
          "max_revives_keep": 71,
          "item_filter": {
            "Pokeball":       { "keep" : 100 },
            "Potion":         { "keep" : 5 },
            "Super Potion":   { "keep" : 10 },
            "Hyper Potion":   { "keep" : 31 },
            "Revive":         { "keep" : 30 },
            "Razz Berry":     { "keep" : 65 }
          }
        }
      },
      {
        "type": "CatchVisiblePokemon"
      },
      {
        "type": "CatchLuredPokemon"
      },
      {
        "type": "SpinFort"
      },
      {
        "type": "MoveToFort",
        "config": {
            "lure_attraction": true,
            "lure_max_distance": 400
        }
      },
      {
        "type": "FollowSpiral",
        "config": {
          "diameter": 4,
          "step_size": 70
        }
      }
    ],
    "map_object_cache_time": 5,
    "forts": {
      "avoid_circles": true,
      "max_circle_size": 3,
      "cache_recent_forts": true
    },
    "websocket_server": false,
    "walk_max": 1.4,
    "walk_min": 0.1,
    "action_wait_min": 5,
    "action_wait_max": 50,
    "debug": false,
    "test": false,
    "health_record": true,
    "location_cache": true,
    "distance_unit": "km",
    "reconnecting_timeout": 83,
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "min_ultraball_to_keep": 5,
    "logging_color": true,
    "catch": {
      "Rattata": {"catch_above_cp": 400, "catch_above_iv": 0.95, "logic": "or"},
      "Pidgey":  {"catch_above_cp": 400, "catch_above_iv": 0.95, "logic": "or"},
      "any":     {"catch_above_cp": 0,   "catch_above_iv": 0,    "logic": "or"},
      "// Example of always catching Rattata:": {},
      "// Rattata": { "always_catch" : true }
    },
    "catch_throw_parameters": {
      "excellent_rate": 0.0,
      "great_rate": 0.1,
      "nice_rate": 0.3,
      "normal_rate": 0.4,
      "spin_success_rate" : 0.2
    },
    "release": {
      "// any": {"release_below_cp": 0, "release_below_iv": 0, "logic": "or"},
      "// Example of always releasing Rattata:": {},
      "// Rattata": {"always_release": true},
      "// Example of keeping 3 stronger (based on CP) Pidgey:": {},
      "// Pidgey": {"keep_best_cp": 3},
      "// Example of keeping 2 stronger (based on IV) Zubat:": {},
      "// Zubat": {"keep_best_iv": 2},
      "// Also, it is working with any": {},
      "// any": {"keep_best_iv": 3},
      "// Example of keeping the 2 strongest (based on CP) and 3 best (based on IV) Zubat:": {},
      "Pidgey":     {"release_below_iv": 0.82, "release_below_cp": 0,   "logic": "or"},
      "Rattata":    {"release_below_iv": 0.89, "release_below_cp": 0,   "logic": "or"},
      "Spearow":    {"release_below_iv": 0.96, "release_below_cp": 0,   "logic": "or"},
      "Weedle":     {"release_below_iv": 0.71, "release_below_cp": 0,   "logic": "or"},
      "Drowzee":    {"release_below_iv": 0.80, "release_below_cp": 0,   "logic": "or"},
      "Hypno":      {"release_below_iv": 0.80, "release_below_cp": 700, "logic": "and"},
      "Aerodacryl": {"release_below_iv": 0.82, "release_below_cp": 839, "logic": "and"},
      "Fearow":     {"release_below_iv": 0.96, "release_below_cp": 814, "logic": "and"},
      "Golduck":    {"release_below_iv": 0.89, "release_below_cp": 806, "logic": "and"},
      "any": {"keep_best_iv": 2}
    },
    "vips" : {
         "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {"catch_above_cp": 1200, "catch_above_iv": 0.9, "logic": "or" },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},

        "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "// Mew evolves to Mewtwo": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {}

    }
}
```

### Output when issue occurred
2016-08-17 12:56:48,542 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:56:48,945 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.06km (attraction of lure 0.06km)
2016-08-17 12:56:50,371 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:56:50,701 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.06km (attraction of lure 0.06km)
2016-08-17 12:56:52,470 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:56:52,827 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.06km (attraction of lure 0.06km)
2016-08-17 12:56:54,661 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:56:55,055 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.06km (attraction of lure 0.06km)
2016-08-17 12:56:57,223 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:56:57,588 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:56:59,678 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:00,079 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:57:02,033 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:02,370 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:57:04,113 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:04,450 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:57:05,919 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:06,304 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:57:08,073 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:08,425 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:57:10,088 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:10,436 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:57:12,066 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:12,406 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.05km (attraction of lure 0.05km)
2016-08-17 12:57:14,240 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:14,582 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.04km (attraction of lure 0.04km)
2016-08-17 12:57:16,207 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:16,695 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.04km (attraction of lure 0.04km)
2016-08-17 12:57:18,520 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:18,870 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.04km (attraction of lure 0.04km)
2016-08-17 12:57:21,295 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:22,049 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.04km (attraction of lure 0.04km)
2016-08-17 12:57:23,691 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:24,031 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Anderer Brunnen am Löwenwall - 0.04km (attraction of lure 0.04km)
2016-08-17 12:57:25,596 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:25,597 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:27,109 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:27,109 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:29,245 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:29,245 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:30,361 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:30,361 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:33,156 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:33,156 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:35,689 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:35,689 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:36,711 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:36,712 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:40,749 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:40,749 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:41,825 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:41,825 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:44,243 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:44,243 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)
2016-08-17 12:57:46,548 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Löwenstatue III (a65ff1016f9a403095476c128d9472e2.11)
2016-08-17 12:57:46,549 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Anderer Brunnen am Löwenwall (3178f031e7bf4c989c86ec1698f60766.16)

### Steps to Reproduce
This happens always as soon as I enable these two lines in my catch config:
```   
      "Rattata": {"catch_above_cp": 400, "catch_above_iv": 0.95, "logic": "or"},
      "Pidgey":  {"catch_above_cp": 400, "catch_above_iv": 0.95, "logic": "or"},
```

### Other Information
OS: OX X
Branch:  master
Git Commit: 
423e4dea72c0b04d8882ef90f03949fe3bf35ac2
Python Version: 
Python 2.7.9

**Comment by k4n30** [comment]
@akanezumi I believe this has been fixed in dev, so unless you want to switch to dev, please wait until it's merged to master

**Comment by tomorrowplusone** [comment]
why did you put it waiting on merge? when has it solved? i use devs and i havent seen a fix for this yet.

**Comment by tomorrowplusone** [comment]
That happens to me too when I use:
```
	  "Rattata": { "never_catch": true },
	  "Spearow": { "never_catch": true },
	  "Pidgey": { "never_catch": true },
	  "Meowth": { "never_catch": true },
	  "Ekans":  { "never_catch": true },
```
i think it does happen when they are around too (without lure), but console doesnt drop any info in that case, just get like frozen.

**Comment by tomorrowplusone** [comment]
@k4n30 **please** remove the "waiting on merge" label, i've just tried it in last dev too.

**Comment by gR8Whl5** [comment]
Same problem here, on current dev branch.  Bot gets stuck looping between lured pokestops when "catch_lured_pokemon" is true, I didn't have any special catch rules activated but it stopped as soon as I changed that to false.

**Comment by k4n30** [comment]
Sorry, must've got mixed up with another issue

**Comment by k4n30** [comment]
Wow failing at labels today :(

**Comment by tomorrowplusone** [comment]
@helloj love you! , thank you man, works like charm.

**Comment by k4n30** [comment]
@helloj Pull request?

**Comment by tomorrowplusone** [comment]
@helloj Pull request?

**Comment by tomorrowplusone** [comment]
@helloj  ye, we apreciate it so much, but why not put the patch in this repo?
there is another issue oppened with the same problem.
